cmake/XrplSanitizers.cmake

```diff
@@ -0,0 +1,174 @@
+#[===================================================================[
+   Configure sanitizers based on environment variables.
+
+   This module reads the following environment variables:
+   - SANITIZERS: The sanitizers to enable. Possible values:
+     - "Address"
+     - "Address,UndefinedBehavior"
+     - "Thread"
+     - "Thread,UndefinedBehavior"
+     - "UndefinedBehavior"
+
+   The compiler type and platform are detected automatically by CMake.
+   The sanitizer compile options are applied to the 'common' interface library
+   which is linked to all targets in the project.
+#]===================================================================]
+
+# Read environment variable
+set(SANITIZERS $ENV{SANITIZERS})
+
+if(NOT SANITIZERS)
+  return()
+endif()
+
+message(STATUS "Configuring sanitizers: ${SANITIZERS}")
+
+# Parse SANITIZERS value to determine which sanitizers to enable
+set(ENABLE_ASAN FALSE)
+set(ENABLE_TSAN FALSE)
+set(ENABLE_UBSAN FALSE)
+
+# Normalize SANITIZERS into a list
+set(_san_list "${SANITIZERS}")
+string(REPLACE "," ";" _san_list "${_san_list}")
+separate_arguments(_san_list)
+
+set(REMAINING_SANITIZERS "")
+foreach(_san IN LISTS _san_list)
+  if(_san STREQUAL "Address")
+    set(ENABLE_ASAN TRUE)
+  elseif(_san STREQUAL "Thread")
+    set(ENABLE_TSAN TRUE)
+  elseif(_san STREQUAL "UndefinedBehavior")
+    set(ENABLE_UBSAN TRUE)
+  else()
+    list(APPEND REMAINING_SANITIZERS "${_san}")
+  endif()
+endforeach()
+
+if(REMAINING_SANITIZERS)
+  message(FATAL_ERROR
+    "Unknown SANITIZERS value(s): ${REMAINING_SANITIZERS}. "
+    "Supported: Address, Thread, UndefinedBehavior and their combinations.")
+endif()
+
+# Detect compiler type
+set(IS_GCC FALSE)
+set(IS_CLANG FALSE)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(IS_GCC TRUE)
+    message(STATUS "  Compiler: GCC ${CMAKE_CXX_COMPILER_VERSION}")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(IS_CLANG TRUE)
+    message(STATUS "  Compiler: Clang ${CMAKE_CXX_COMPILER_VERSION}")
+endif()
+
+# Detect platform (amd64/x86_64 vs arm64/aarch64)
+set(IS_AMD64 FALSE)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|AMD64")
+    set(IS_AMD64 TRUE)
+    message(STATUS "  Platform: amd64")
+else()
+    message(STATUS "  Platform: ${CMAKE_SYSTEM_PROCESSOR}")
+endif()
+
+# Frame pointer is required for meaningful stack traces. Sanitizers recommend minimum of -O1 for reasonable performance
+set(SANITIZERS_COMPILE_FLAGS "-fno-omit-frame-pointer" "-O1")
+
+# Build the sanitizer flags list
+set(SANITIZERS_FLAGS)
+
+if(ENABLE_ASAN)
+    list(APPEND SANITIZERS_FLAGS "address")
+elseif(ENABLE_TSAN)
+    list(APPEND SANITIZERS_FLAGS "thread")
+endif()
+
+if(ENABLE_UBSAN)
+    # UB sanitizer flags
+    if(IS_CLANG)
+        # Clang supports additional UB checks
```

Comment:
The additional flags you added is not all the sanitizers clang supports when running with `UndefinedBehaviorSanitizer`

https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html

> -fsanitize=undefined: All of the checks listed above other than float-divide-by-zero, unsigned-integer-overflow, implicit-conversion, local-bounds, vptr and the nullability-* group of checks.

Maybe we should:
- add a link to docs
- either enable all additional flags, all comment why we didn't add them